File: PYTHON_SOLUTIONS/11.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        l,r = 0, len(height) - 1
        res = 0
        while l < r:
            mh = min(height[l], height[r])
            res = max(res, (r - l) * mh)
            while(l < r and height[l] <= mh):
                l += 1
            while(r > l and height[r] <= mh):
                r -= 1
        return res

File: CPP_SOLUTIONS/11.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0, r = height.size() - 1;
        int maxwater = 0;
        while(l < r) {
            int mh = min(height[l], height[r]);
            maxwater = max(maxwater, (r - l) * mh);
            if(mh == height[l]) {
                while(l < r && height[l] <= mh) l++;
            } else {
                while(l < r && height[r] <= mh) r--;
            }
        }
        return maxwater;
        
    }
};
